astropy/io/registry.py

```diff
@@ -321,9 +322,9 @@ def read(cls, *args, **kwargs):
             if len(args):
                 if isinstance(args[0], PATH_TYPES):
                     from ..utils.data import get_readable_fileobj
-                    # For Py3 the path might be a pathlib.Path object, so coerce
-                    # to a regular string.
-                    if six.PY3:
+                    # path might be a pathlib.Path object if HAS_PATHLIB,
+                    # so coerce to a regular string.
+                    if HAS_PATHLIB:
```

Comment:
It occurs to me that in theory a user might be passing a unicode file name on Python 2 with the backport of pathlib independently installed.  Then conversion with `str()` would be a problem.  So maybe you need to also check `isinstance(args[0], pathlib.Path)`?